File: snippets/kotlin/src/main/kotlin/Relay.kt
package rust.nostr.snippets

import kotlinx.coroutines.runBlocking
import org.nostrdevkit.sdk.LocalRelayBuilder
import org.nostrdevkit.sdk.NostrLmdb

suspend fun runEmbeddedRelay() {
    val database = NostrLmdb.open("./data/relay")
    val relay = LocalRelayBuilder()
        .database(database)
        .port(7777u)
        .build()

    relay.run()
    println("Relay URL: ${relay.url()}")

}

fun relayMain() = runBlocking {
    runEmbeddedRelay()
}

File: snippets/js/node/src/relay.ts
import { LocalRelayBuilder, NostrLmdb } from "@nostrdevkit/nostr-sdk-node";

const database = await NostrLmdb.open("./data/relay");
const relay = new LocalRelayBuilder()
    .database(database)
    .port(7777)
    .build();

await relay.run();
console.log("Relay URL:", (await relay.url()).toString());
